src/test/java/org/openrewrite/java/micronaut/UpdateMavenAnnotationProcessorsTest.java

```diff
@@ -0,0 +1,333 @@
+                        <scope>compile</scope>
+                    </dependency>
+                    <dependency>
+                        <groupId>ch.qos.logback</groupId>
+                        <artifactId>logback-classic</artifactId>
+                        <scope>runtime</scope>
+                    </dependency>
+                </dependencies>
+                <build>
+                    <plugins>
+                        <plugin>
+                            <groupId>io.micronaut.maven</groupId>
+                            <artifactId>micronaut-maven-plugin</artifactId>
+                        </plugin>
+                        <plugin>
+                            <groupId>org.apache.maven.plugins</groupId>
+                            <artifactId>maven-compiler-plugin</artifactId>
+                            <configuration>
+                                <annotationProcessorPaths combine.children="append">
+                                    <path>
+                                        <groupId>io.micronaut.data</groupId>
+                                        <artifactId>micronaut-data-processor</artifactId>
+                                        <version>${micronaut.data.version}</version>
+                                    </path>       
+                                    <path>
+                                        <groupId>io.micronaut</groupId>
+                                        <artifactId>micronaut-inject-java</artifactId>
+                                        <version>${micronaut.core.version}</version>
+                                    </path>
+                                    <path>
+                                        <groupId>io.micronaut</groupId>
+                                        <artifactId>micronaut-http-validation</artifactId>
+                                        <version>${micronaut.core.version}</version>
+                                    </path>
+                                    <path>
+                                        <groupId>io.micronaut</groupId>
+                                        <artifactId>micronaut-graal</artifactId>
+                                        <version>${micronaut.core.version}</version>
+                                    </path>
+                                </annotationProcessorPaths>
+                                <compilerArgs>
+                                    <arg>-Amicronaut.processing.group=com.example</arg>
+                                    <arg>-Amicronaut.processing.module=demo</arg>
+                                </compilerArgs>
+                            </configuration>
+                        </plugin>
+                    </plugins>
+                </build>
+            </project>
+            """.formatted(latestMicronautVersion))));
+    }
+
+    @Test
+    void updateCoreMavenAnnotationProcessorsAndDontModifyModuleProcessors() {
```

Comment:
Would a parameterized test help to reduce the duplication here?

Reply:
Possibly...but putting the globs of XML in the parameters seems perhaps not worth the extra ceremony when it's just a couple of cases.